aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2014,7 +2017,7 @@ static Tensor _matmul_impl(
                    : tensor1.unsqueeze(0).mm(tensor2).squeeze_(0);
   } else if (dim_tensor1 == 2 && dim_tensor2 == 2) {
     return has_out ? at::mm_out(out, tensor1, tensor2) : tensor1.mm(tensor2);
-  } else if (should_fold(tensor1, tensor2)) {
+  } else if (!has_out && should_fold(tensor1, tensor2)) {
```

Comment:
If we go with this fix, there is a bit of code in this branch that's now dead.

Reply:
Good catch. But now that I think about it, I actually prefer the previous version for a smaller blast radius lol